### ISSUE TITLE
Lost 2 VM's on mainnet while chain contract state is "Created"

### ISSUE THREAD

**Issue by coesensbert** [threefoldtech/zos#1694]
Deployed big BSC full node VM on ZOS node 306 (GE), let it sync overnight while it suddenly became unavailable the next night at 3:28 CEST.

Terraform:
```
terraform {
  required_providers {
    grid = {
      source = "threefoldtech/grid"
    }
  }
}

provider "grid" {
}

resource "grid_network" "net1" {
    nodes = [306]
    ip_range = "10.138.0.0/16"
    name = "bscnodenetw"
    description = "BSC full node network"
    add_wg_access = true
}
resource "grid_deployment" "d1" {
  node = 306
  network_name = grid_network.net1.name
  ip_range = lookup(grid_network.net1.nodes_ip_range, 306, "")
  disks {
    name = "root"
    size = 1700
  }
    vms {
    name = "bscfullnode"
    description ="Binance Smart Chain full node"
    flist = "https://hub.grid.tf/tf-official-vms/ubuntu-20.04-lts.flist"
    cpu = 32
    publicip = true
    publicip6 = false
    memory = 65536
#    rootfs_size = 25000
    mounts {
        disk_name = "root"
        mount_point = "/data"
    }
    planetary = true
    env_vars = {
      SSH_KEY ="ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIDYNeJXJV2FNEwuQz6e0jkKeqRbKWwftBKq+sjSTqa2x"
    }
  }
}
```

Terrafor show:
```
➜  bsc-node git:(master) ✗ terraform show       
# grid_deployment.d1:
resource "grid_deployment" "d1" {
    id           = "1351"
    ip_range     = "10.138.3.0/24"
    network_name = "bscnodenetw"
    node         = 306

    disks {
        name = "root"
        size = 1700
    }

    vms {
        computedip  = "45.156.243.132/25"
        cpu         = 32
        description = "Binance Smart Chain full node"
        env_vars    = {
            "SSH_KEY" = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIDYNeJXJV2FNEwuQz6e0jkKeqRbKWwftBKq+sjSTqa2x"
        }
        flist       = "https://hub.grid.tf/tf-official-vms/ubuntu-20.04-lts.flist"
        ip          = "10.138.3.2"
        memory      = 65536
        name        = "bscfullnode"
        planetary   = true
        publicip    = true
        publicip6   = false
        rootfs_size = 0
        ygg_ip      = "300:14f4:8064:f728:6302:20c7:7331:d5cd"

        mounts {
            disk_name   = "root"
            mount_point = "/data"
        }
    }
}

# grid_network.net1:
resource "grid_network" "net1" {
    access_wg_config   = <<-EOT
        
        [Interface]
        Address = 100.64.138.2
        PrivateKey = oBo4zd5+t2CcAaiaYRtT3oQAsoV+bL7pRYkGAAKKRG4=
        [Peer]
        PublicKey = ly5ka6C3ldf+nAnbp9SQZC9kWXfibXYP5gx6cwtKvnY=
        AllowedIPs = 10.138.0.0/16, 100.64.0.0/16
        PersistentKeepalive = 25
        Endpoint = 45.156.243.246:3367
        
    EOT
    add_wg_access      = true
    description        = "BSC full node network"
    external_ip        = "10.138.2.0/24"
    external_sk        = "oBo4zd5+t2CcAaiaYRtT3oQAsoV+bL7pRYkGAAKKRG4="
    id                 = "ebaad94c-dc71-48ab-9f9e-d61f11697dc4"
    ip_range           = "10.138.0.0/16"
    name               = "bscnodenetw"
    node_deployment_id = {
        "306" = 1348
    }
    nodes              = [
        306,
    ]
    nodes_ip_range     = {
        "306" = "10.138.3.0/24"
    }
    public_node_id     = 306
}


Outputs:

node1_vm1_ip = "10.138.3.2"
public_ip = "45.156.243.132/25"
public_ip6 = ""
wg_config = <<-EOT
    
    [Interface]
    Address = 100.64.138.2
    PrivateKey = oBo4zd5+t2CcAaiaYRtT3oQAsoV+bL7pRYkGAAKKRG4=
    [Peer]
    PublicKey = ly5ka6C3ldf+nAnbp9SQZC9kWXfibXYP5gx6cwtKvnY=
    AllowedIPs = 10.138.0.0/16, 100.64.0.0/16
    PersistentKeepalive = 25
    Endpoint = 45.156.243.246:3367
    
EOT
ygg_ip = "300:14f4:8064:f728:6302:20c7:7331:d5cd"
```

On chain contract is in state enabled: https://graph.grid.tf/graphql

```
query MyQuery {
  nodeContracts(where: {contractID_eq: "1351"}) {
    id
    state
    numberOfPublicIPs
    twinID
    nodeID
    createdAt
  }
}
```

Loki logs for node 306: https://mon.grid.tf/explore?orgId=1&left=%5B%221652306728790%22,%221652344671490%22,%22Loki%22,%7B%22refId%22:%22A%22,%22expr%22:%22%7Bnetwork%3D%5C%22production%5C%22,node%3D%5C%22306%5C%22%7D%22%7D%5D

Same scenario happened this night with another vm on ZOS node 315
VM contract ID: 1385
Net contract ID: 1383 + 1384

**Comment by muhamadazmy** [comment]
I am so sorry for your loss. I started an investigation on what can caused this. Unfortunately duo to a bug the `reason` of the deletion was not stored correctly so I am not sure what triggered a full deployment deletion.

Deletion of the entire deployment can happen for one the following reason
- Contract cancel even received from the chain
- API call to delete the contract but this has to be done with the contract owner keys so, unless you have the twin keys it's not possible. I also disabled this (as part of a fix for this issue) to make rule it out.
- Node synchronizing active contracts with the chain and it found that the contract is not in the active node contracts list.

The only thing we can do now is to redeploy (possibly on the same node) and wait until this happens again. This way we should have enough insight on what actually triggered the deletion.
